cmd/zapi/cmd/info/ls.go

```diff
@@ -46,33 +45,37 @@ func (c *LsCommand) Run(args []string) error {
 	matches, err := cmd.SpaceGlob(c.Context(), conn, args...)
 	if err != nil {
 		if err == cmd.ErrNoSpacesExist {
-			fmt.Println("no spaces exist")
```

Comment:
I think this is a nice to have, maybe not printed to stdout though.

Reply:
talked with matt, said that the lack of output if no spaces were present matched `ls`, he agreed.

Reply:
Sorry to chime in late on what might be a settled topic, but I lean more toward what @mattnibs was implying. If a user has typed what's effectively a "show me something" command, and there's nothing to show, I've generally always been in favor of an explicit "there is nothing to show" sort of response, rather than silence being used to implicitly communicate this. I've always leaned this way because "early exit" bugs can happen where the there _is_ data that should have been shown but the tools just silently failed. So the explicit response helps differentiate between the two.

Also, I agree that stdout would not be the appropriate place for the message in any case, since a user piping might be piping the output to something downstream.

All that said, none of this is a hill I'm prepared to die on, so I'm also prepared to be overruled.

Reply:
I think we should mirror what `ls` does, similar to other unix like tools, where it's acceptable to return exit status of 0 and nothing to stdout on success, so I'm going to merge in as-is.

We already have this behavior today for zq, btw, where you can execute zq and get silent success, because no records were returned or generated.